src/service.c

```diff
@@ -655,9 +655,22 @@ create_switch_section (IndicatorSessionService * self, int profile)
     {
       if (want_accel)
         {
-          gchar * str = g_settings_get_string (p->keybinding_settings, "screensaver");
-          g_menu_item_set_attribute (item, "accel", "s", str);
-          g_free (str);
+            if (ayatana_common_utils_is_mate())
```

Comment:
Does this really only apply for MATE? What about Budgie, GNOME (in theory), Unity7? see line 1255 in this file before this change has been applied.

Reply:
> Does this really only apply for MATE? What about Budgie, GNOME (in theory), Unity7? see line 1255 in this file before this change has been applied.

Only MATE stores this as a string, all other desktops use `org.gnome.settings-daemon.plugins.media-keys`, which stores this value as a list.